lightning/src/chain/chaininterface.rs

```diff
@@ -376,3 +386,59 @@ impl ChainWatchInterfaceUtil {
 		watched.does_match_tx(tx)
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use ln::functional_test_utils::{create_node_cfgs};
+	use super::{BlockNotifier, ChainListener};
+	use std::ptr::eq as ptr_eq;
+
+	#[test]
+	fn register_listener_test() {
+		let node_cfgs = create_node_cfgs(1);
+		let block_notifier = BlockNotifier::new(node_cfgs[0].chain_monitor.clone());
+		let listener = &node_cfgs[0].chan_monitor.simple_monitor as &ChainListener;
+		block_notifier.register_listener(listener);
+		let vec = block_notifier.listeners.lock().unwrap();
+		assert_eq!(vec.len(), 1);
+		let item = vec.first().clone().unwrap();
```

Comment:
Can you add a test that you can deregister node_cfgs[0].chan_monitor.simple_monitor itself instead of just the thing that was pulled out of the registration list?

Reply:
I'm not sure I fully understand what you mean by 

> instead of just the thing that was pulled out of the registration list

Do you mean calling `unregister_listener` and passing in `node_cfgs[0].chan_monitor.simple_monitor` itself instead of the `listener1` local variable?

If so, I wrote a new test `unregister_single_listener_ref_test`, otherwise would love some guidance.

Reply:
Thats exactly what I meant.